### PR TITLE
SPE-746: basic auth fix

### PR DESCRIPTION
**Pull request by aleksandr-nesterov**

### CODE REVIEW

File: simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/component/securityctrlr/BasicAuthPasswordVariableAccessor.java [hunk 1]
```diff
@@ -98,14 +98,18 @@ private SetVariableResult validateActualValue(AttributePath attributePath, CiStr
...
+        if (invalidLength(attributeValue) || isNotHex(attributeValue.toString()) || isEven(attributeValue)) {
             return setVariableResult.withAttributeStatus(SetVariableResult.AttributeStatus.INVALID_VALUE);
         }
 
         return setVariableResult.withAttributeStatus(SetVariableResult.AttributeStatus.ACCEPTED);
 
     }
 
+    private boolean isEven(CiString.CiString1000 attributeValue) {
+        return (attributeValue.toString().length() & 0x1) == 1;
```
Comment (draganevbox) [inline simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/component/securityctrlr/BasicAuthPasswordVariableAccessor.java]: 👍

Comment (akoshterek) [reply simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/component/securityctrlr/BasicAuthPasswordVariableAccessor.java]: lol. it's isOdd actually. 
For even numbers length() & 0x1 == 0

Comment (akoshterek) [reply simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/component/securityctrlr/BasicAuthPasswordVariableAccessor.java]: Even: 2, 4, 50, ..
Odd 1,3, 5, 7,...

Comment (aleksandr-nesterov) [reply simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/component/securityctrlr/BasicAuthPasswordVariableAccessor.java]: yes,  s/isEven/isOdd

**Pull request merged**